scipy/ndimage/tests/test_filters.py

```diff
@@ -2016,6 +2016,38 @@ def test_rank15(self, dtype, xp):
                                      origin=[-1, 0])
         assert_array_almost_equal(expected, output)
 
+    def test_rank16(self):
+        array = [3, 2, 5, 1, 4]
+        expected = np.asarray([3, 3, 2, 4, 4])
```

Comment:
Here and elsewhere:

```suggestion
        expected = xp.asarray([3, 3, 2, 4, 4])
```

We now allow inputs not only numpy arrays, and make sure that namespace out == namespace in. And `xp` parametrization is over the array api compliant libraries (currently, torch, jax.numpy, cupy)

Reply:
I think that xp_assert_equal covers it, right? I have converted everything to it...

Reply:
`xp_assert_equal(actual, desired)` checks that `actual` and `desired` have the same namespace, so it'll error out if actual is a torch tensor and expected is a numpy array.